vllm/v1/engine/core.py

```diff
@@ -203,7 +236,21 @@ def step(self) -> EngineCoreOutputs:
                 scheduler_stats=self.scheduler.make_stats(),
             )
         scheduler_output = self.scheduler.schedule()
-        output = self.model_executor.execute_model(scheduler_output)
+        try:
+            output = self.model_executor.execute_model(scheduler_output)
+        except BaseException as err:
+            # NOTE: ensure we can log extra info without risking raises
+            # unexpected errors during logging
+            with contextlib.suppress(BaseException):
+                model_err = ModelExecutionError(
+                    f"Model execution failure,"
+                    f"reason: {repr(err)}",
+                    scheduler_output=scheduler_output,
+                    scheduler_stats=self.scheduler.make_stats())
+                dump_engine_exception(model_err, self.vllm_config)
+            # Re-raise exception
+            raise err
```

Comment:
Would it be possible to move this out of the `step` method so as not to impact readability? Can put all this in another method that's called from here, e.g.
```python
def execute_model(self, scheduler_output: SchedulerOutput):
    try:
        output = self.model_executor.execute_model(scheduler_output)
    except BaseException as err:
        # ...
```